core/etcd/config/templates/stack-template.json

```diff
@@ -451,7 +451,7 @@
       },
       "DependsOn": [
         {{if $etcdInstance.DependencyExists}}{{$etcdInstance.DependencyRef}},{{end}}
-        {{if $etcdIndex}}"{{$.Etcd.LogicalName}}{{sub $etcdIndex 1}}",{{end}}
+        {{if $.Etcd.StackExists}}{{if $etcdIndex}}"{{$.Etcd.LogicalName}}{{sub $etcdIndex 1}}",{{end}}{{end}}
```

Comment:
Can you make it just `{{ if .IsUpdatingCluster }}` and pass a boolean value `true` for `kube-aws update`, and `false` for `kube-aws up`? It would simplify the implementation.

Reply:
Yes, I was tempted to re-factor this in a similar way but in the end resisted for the following reasons: -

* We can build upon this implementation in order to build to proposed fail-fast when someone tries to update a legacy style implementation to the separate stacks implementation which will break their cluster.  The other simpler way does not allow this.
* By implementing the utility StackExists/NestedStackExists functions we can make use of these to implement other useful UI simplifications, such as creating a kube-aws 'apply' command that handles cfn create and update for us and would make implementing kube-aws into pipelines easier.

What do you think?  Is it worth sacrificing the simplicity now for these other features?  I think I could PR the fail-fast in the next day or two and the 'apply' command wouldn't take much longer?

Reply:
Context: https://github.com/kubernetes-incubator/kube-aws/pull/1357#discussion_r194323349

Reply:
> this implementation in order to build to proposed fail-fast when someone tries to update a legacy style implementation to the separate stacks implementation which will break their cluster

Ah, makes sense!

> Is it worth sacrificing the simplicity now for these other features?

Definitely! It turned out that this isn't sacrificing anything, given two use-cases of the `kube-aws apply` and the fail-fast feature as context.

Thank you so much for the clarifications 😄